src/project.js

```diff
@@ -185,6 +200,24 @@ function createVisibilityMatrix (result) {
     result.changeCost = percentifyDensity(changeCost, visibilityMatrix);
 }
 
+// where we have 0, set distance to Infinity and copy the matrix so its special
```

Comment:
I'd move this comment to just above the `|| Infinity` line where the assignment is.

Also, I'm not sure I understand what you mean by "...copy the matrix so its special"?